packages/astro/src/core/build/plugins/util.ts

```diff
@@ -54,10 +54,12 @@ export function getVirtualModulePageNameFromPath(virtualModulePrefix: string, pa
 	// we mask the extension, so this virtual file
 	// so rollup won't trigger other plugins in the process
 	const extension = extname(path);
-	return `${virtualModulePrefix}${path.replace(
-		extension,
-		extension.replace('.', ASTRO_PAGE_EXTENSION_POST_PATTERN)
-	)}`;
+	return (
+		virtualModulePrefix +
+		(extension === '.astro'
+			? path.slice(0, -extension.length) + `${ASTRO_PAGE_EXTENSION_POST_PATTERN}astro`
+			: path)
+	);
```

Comment:
Can you update the documentation of the function to take in account the new logic?